libc/string_errsig.c

```diff
@@ -0,0 +1,247 @@
+#else
+static const int error_codes[] = { E2BIG, EACCES, EADDRINUSE, EADDRNOTAVAIL, EAFNOSUPPORT, EAGAIN, EALREADY, EBADF, EBADMSG, EBUSY, ECANCELED, ECHILD, ECONNABORTED, ECONNREFUSED, ECONNRESET, EDEADLK, EDESTADDRREQ, EDOM, EDQUOT, EEXIST, EFAULT, EFBIG, EHOSTUNREACH, EIDRM, EILSEQ, EINPROGRESS, EINTR, EINVAL, EIO, EISCONN, EISDIR, ELOOP, EMFILE, EMLINK, EMSGSIZE, EMULTIHOP, ENAMETOOLONG, ENETDOWN, ENETRESET, ENETUNREACH, ENFILE, ENOBUFS, ENODATA, ENODEV, ENOENT, ENOEXEC, ENOLCK, ENOLINK, ENOMEM, ENOMSG, ENOPROTOOPT, ENOSPC, ENOSR, ENOSTR, ENOSYS, ENOTCONN, ENOTDIR, ENOTEMPTY, ENOTRECOVERABLE, ENOTSOCK, ENOTSUP, ENOTTY, ENXIO, EOPNOTSUPP, EOVERFLOW, EOWNERDEAD, EPERM, EPIPE, EPROTO, EPROTONOSUPPORT, EPROTOTYPE, ERANGE, EROFS, ESPIPE, ESRCH, ESTALE, ETIME, ETIMEDOUT, ETXTBSY, EWOULDBLOCK, EXDEV, ENOTBLK };
+#endif
+
+
+TEST_GROUP(string_errsingn);
+
+
+TEST_SETUP(string_errsingn)
+{
+}
+
+
+TEST_TEAR_DOWN(string_errsingn)
+{
+}
+
+
+TEST(string_errsingn, strerror_basic)
+{
+	for (int i = 0; i < (sizeof error_codes / sizeof *error_codes); i++) {
+		errno = 0;
+		TEST_ASSERT_NOT_NULL(strerror(error_codes[i - 1]));
+		TEST_ASSERT_NOT_NULL(strerror(error_codes[i]));
+		TEST_ASSERT_LESS_OR_EQUAL_INT(MAX_LEN_STRING, strlen(strerror(error_codes[i - 1])));
+		TEST_ASSERT_LESS_OR_EQUAL_INT(MAX_LEN_STRING, strlen(strerror(error_codes[i])));
+		TEST_ASSERT_TRUE(strerror(error_codes[i - 1]) != strerror(error_codes[i]));
+		TEST_ASSERT_EQUAL_INT(0, errno);
+	}
+}
+
+
+TEST(string_errsingn, strerror_zero)
+{
+	errno = 0;
+	TEST_ASSERT_NOT_NULL(strerror(0));
+	TEST_ASSERT_LESS_OR_EQUAL_INT(MAX_LEN_STRING, strlen(strerror(0)));
+	TEST_ASSERT_EQUAL_INT(0, errno);
+}
+
+
+TEST(string_errsingn, strerror_invaild)
+{
+	int values[] = { INT_MIN, INT_MIN / 2, -1024, 1024, INT_MAX / 2, INT_MAX };
+	for (int i = 0; i < (sizeof values / sizeof *values); i++) {
+		TEST_ASSERT_NOT_NULL(strerror(values[i]));
+		TEST_ASSERT_NOT_NULL(strstr(strerror(values[i]), "Unknown error"));
+		TEST_ASSERT_LESS_OR_EQUAL_INT(MAX_LEN_STRING, strlen(strerror(values[i])));
+	}
+}
+
+
+TEST(string_errsingn, strerror_r_basic)
+{
+/* strerror_r() not yet implemented in Phoenix-RTOS */
+#ifdef __phoenix__
+	TEST_IGNORE();
+#else
+	for (int i = 0; i < (sizeof error_codes / sizeof *error_codes); i++) {
+		char prev_buf[MAX_LEN_STRING] = { 0 };
+		char buf[MAX_LEN_STRING] = { 0 };
+
+		errno = 0;
+		TEST_ASSERT_EQUAL_INT(0, strerror_r(error_codes[i - 1], prev_buf, sizeof(buf)));
+		TEST_ASSERT_EQUAL_INT(0, strerror_r(error_codes[i], buf, sizeof(buf)));
+		TEST_ASSERT_LESS_OR_EQUAL_INT(MAX_LEN_STRING, strlen(prev_buf));
+		TEST_ASSERT_LESS_OR_EQUAL_INT(MAX_LEN_STRING, strlen(buf));
+		TEST_ASSERT_TRUE(buf != prev_buf);
```

Comment:
always true, You're checking pointers for match, not the buffer contents pointers are pointing to